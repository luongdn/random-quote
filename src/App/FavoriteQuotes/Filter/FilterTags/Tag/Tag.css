.filter__tag {
  display: inline-block;
  margin: auto 5px;
  cursor: pointer;
}

.filter_tag--selected {
  font-weight: bold;
}