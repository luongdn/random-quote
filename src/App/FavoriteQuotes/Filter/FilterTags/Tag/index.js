import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './Tag.css'

class Tag extends Component {
  render() {
    const { onToggle, name, selected } = this.props

    return(
      <div
        className="filter__tag"
        onClick={onToggle}
      >
        <p className={selected ? 'filter_tag--selected' : ''}>{name}</p>
      </div>
    )
  }
}

Tag.propTypes = {
  onToggle: PropTypes.func,
  name: PropTypes.string,
  selected: PropTypes.bool
}

export default Tag
