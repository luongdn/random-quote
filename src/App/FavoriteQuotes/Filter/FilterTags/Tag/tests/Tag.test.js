import React from 'react'
import Tag from '../'

const setup = (props) => {
  const component = shallow(
    <Tag {...props} />
  )

  return {
    component
  }
}

describe('Tag component', () => {
  it('should match snapshot when tag is selected', () => {
    const { component } = setup({ name: 'tag name', selected: true })
    expect(component).toMatchSnapshot()
  })

  it('should match snapshot when tag is not selected', () => {
    const { component } = setup({ name: 'tag name', selected: false })
    expect(component).toMatchSnapshot()
  })
})
