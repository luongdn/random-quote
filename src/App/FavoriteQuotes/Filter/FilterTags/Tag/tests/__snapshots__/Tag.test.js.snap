// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component should match snapshot when tag is not selected 1`] = `
<div
  className="filter__tag"
>
  <p
    className=""
  >
    tag name
  </p>
</div>
`;

exports[`Tag component should match snapshot when tag is selected 1`] = `
<div
  className="filter__tag"
>
  <p
    className="filter_tag--selected"
  >
    tag name
  </p>
</div>
`;
